### ISSUE TITLE
Settings Issues

### ISSUE THREAD

**Issue by umardraz** [biosistemika/scinote-web#13]
Hello,

1 = During installation of scinote I had used localhost as a mail server in application.yml. I don't think so there is any email server running on scinote-web container. 

How I can change the email server setting after scinote installation?

2 = Is there any way I can install this app on a server instead of Docker?

Br.

Umar

**Comment by jbargu** [comment]
1) Just change environment variables in application.yml used for mail server to your values, namely SMTP* variables and restart server. See [here](https://github.com/biosistemika/scinote-web#user-content-environmental-variables).

2) I suppose you could only install rails and run it from there. But if you can do that, you can surely install docker as well and run it inside - much simpler and pain free. If you choose different option, we might not be able to help you. sciNote is tested in given docker environment and might not work in different environment. As a suggestion deploying to Heroku is pretty [straightforward](https://github.com/biosistemika/scinote-web#deploy-onto-heroku).

**Comment by umardraz** [comment]
Hello here i am bit confused i have user 1000 where I have folder sinote-web and also application.yml file is there in config folder,

Now i need to change that file or I need to enter into running sicnoteweb_web_1 container and edit that application.yml there?

Br
Umar

**Comment by jbargu** [comment]
Imagine docker container as operating system, which has scinote-web folder mounted as a disk. Therefore if you change something on host, it will be reflected in docker container. Because it's config file you have to restart server though, because variables are loaded at the start. If you enter container you'll find out that it's difficult to change files since no editor is installed (it contains bare bones to run rails app).

On the other hand, you cannot add programs in container. For example if you run `apt-get install <program>` it will probably work in current session, but when you'll restart, it will be gone, since container is static.

In short, enter container when you need to something with rails console, rake tasks or database, because these programs are already installed (specified in Dockerfile) in container for that purpose (basically administration). For editing, deleting, adding files you don't need to enter container.

**Comment by umardraz** [comment]
ok cleared.

I have changed the mail server settings and restart the containers, after that tried to signup the user but there is no logs in my mail server.

Here is the logs of development.log

[2016-06-03 10:09:52 +0000] DEBUG:
[2016-06-03 10:09:52 +0000] DEBUG:
[2016-06-03 10:09:52 +0000] INFO: Started POST "/users" for 182.190.240.11 at 2016-06-03 10:09:52 +0000
[2016-06-03 10:09:52 +0000] INFO: Cannot render console from 182.190.240.11! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
[2016-06-03 10:09:52 +0000] INFO: Processing by Users::RegistrationsController#create as HTML
[2016-06-03 10:09:52 +0000] INFO:   Parameters: {"utf8"=>"✓", "authenticity_token"=>"VrH0xRf9as1M9LWR/2E4SJ1tOGlVDl2igJJO1YAApr2UHnmTrRZD7RBr07kIB6+3ZOxLhhZGHcyi+co4Lx9nMQ==", "user"=>{"full_name"=>"Umar Draz", "email"=>"umar.draz@umardraz.com", "password"=>"[FILTERED]", "password_confirmation"=>"[FILTERED]"}, "organization"=>{"name"=>"UmarDraz"}, "commit"=>"Sign up"}
[2016-06-03 10:09:53 +0000] DEBUG:   User Exists (0.6ms)  SELECT  1 AS one FROM "users" WHERE "users"."email" = 'umar.draz@umardraz.com' LIMIT 1
[2016-06-03 10:09:53 +0000] DEBUG:    (0.3ms)  BEGIN
[2016-06-03 10:09:53 +0000] DEBUG:   User Exists (0.5ms)  SELECT  1 AS one FROM "users" WHERE "users"."email" = 'umar.draz@umardraz.com' LIMIT 1
[2016-06-03 10:09:53 +0000] DEBUG:   User Load (0.4ms)  SELECT  "users".* FROM "users" WHERE "users"."confirmation_token" = $1  ORDER BY "users"."id" ASC LIMIT 1  [["confirmation_token", "b152712492a1937ad969c0f5b361a3424f4440fe644f71ece2fd5ead0f9b02aa"]]
[2016-06-03 10:09:53 +0000] DEBUG:   SQL (0.5ms)  INSERT INTO "users" ("full_name", "email", "encrypted_password", "initials", "created_at", "updated_at", "confirmation_token", "confirmation_sent_at") VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING "id"  [["full_name", "Umar Draz"], ["email", "umar.draz@umardraz.com"], ["encrypted_password", "$2a$10$amnrXrwzChXT8akWrky6UuzghQMalxGmNMAkXN.sEhxqguCzeqOTa"], ["initials", "UD"], ["created_at", "2016-06-03 10:09:53.132874"], ["updated_at", "2016-06-03 10:09:53.132874"], ["confirmation_token", "ByDsvBnN1u4PG3YifQsb"], ["confirmation_sent_at", "2016-06-03 10:09:53.134330"]]
[2016-06-03 10:09:53 +0000] DEBUG:    (0.8ms)  COMMIT
[2016-06-03 10:09:53 +0000] DEBUG:    (0.2ms)  BEGIN
[2016-06-03 10:09:53 +0000] DEBUG:   SQL (0.3ms)  INSERT INTO "delayed_jobs" ("handler", "queue", "priority", "run_at", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5, $6) RETURNING "id"  [["handler", "--- !ruby/object:Delayed::PerformableMethod\nobject: !ruby/object:Devise::Async::Backend::DelayedJob {}\nmethod_name: :perform\nargs:\n- :confirmation_instructions\n- User\n- '39'\n- ByDsvBnN1u4PG3YifQsb\n- {}\n"], ["queue", "devise_email"], ["priority", 10], ["run_at", "2016-06-03 10:09:53.138775"], ["created_at", "2016-06-03 10:09:53.138863"], ["updated_at", "2016-06-03 10:09:53.138863"]]
[2016-06-03 10:09:53 +0000] DEBUG:    (0.5ms)  COMMIT
[2016-06-03 10:09:53 +0000] DEBUG:    (0.2ms)  BEGIN
[2016-06-03 10:09:53 +0000] DEBUG:   SQL (0.4ms)  INSERT INTO "organizations" ("name", "created_by_id", "created_at", "updated_at") VALUES ($1, $2, $3, $4) RETURNING "id"  [["name", "UmarDraz"], ["created_by_id", 39], ["created_at", "2016-06-03 10:09:53.142086"], ["updated_at", "2016-06-03 10:09:53.142086"]]
[2016-06-03 10:09:53 +0000] DEBUG:    (0.6ms)  COMMIT
[2016-06-03 10:09:53 +0000] DEBUG:    (0.1ms)  BEGIN
[2016-06-03 10:09:53 +0000] DEBUG:   SQL (0.5ms)  INSERT INTO "user_organizations" ("user_id", "organization_id", "role", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5) RETURNING "id"  [["user_id", 39], ["organization_id", 40], ["role", 2], ["created_at", "2016-06-03 10:09:53.145178"], ["updated_at", "2016-06-03 10:09:53.145178"]]
[2016-06-03 10:09:53 +0000] DEBUG:    (0.5ms)  COMMIT
[2016-06-03 10:09:53 +0000] INFO: Redirected to http://scinote.mydomain.net/users/sign_in
[2016-06-03 10:09:53 +0000] INFO: Completed 302 Found in 187ms (ActiveRecord: 6.2ms)
[2016-06-03 10:09:53 +0000] DEBUG:
[2016-06-03 10:09:53 +0000] DEBUG:
[2016-06-03 10:09:53 +0000] INFO: Started GET "/users/sign_in" for 182.190.240.11 at 2016-06-03 10:09:53 +0000
[2016-06-03 10:09:53 +0000] INFO: Cannot render console from 182.190.240.11! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
[2016-06-03 10:09:53 +0000] INFO: Processing by Users::SessionsController#new as HTML
[2016-06-03 10:09:53 +0000] INFO:   Rendered devise/shared/_links.html.erb (3.3ms)
[2016-06-03 10:09:53 +0000] INFO:   Rendered devise/sessions/new.html.erb within layouts/main (7.4ms)
[2016-06-03 10:09:54 +0000] INFO:   Rendered shared/_navigation.html.erb (8.2ms)
[2016-06-03 10:09:54 +0000] INFO:   Rendered layouts/application.html.erb (741.6ms)
[2016-06-03 10:09:54 +0000] INFO: Completed 200 OK in 753ms (Views: 751.6ms | ActiveRecord: 0.0ms)

**Comment by jbargu** [comment]
Please read through `README.md` [first](https://github.com/biosistemika/scinote-web#delayed-jobs). You have to run `rake jobs:work` in container.

**Comment by umardraz** [comment]
Ok Thanks

Now what is the wrong with these ?

scinote@d23f4311f1e8:/usr/src/app$ rake data:clean_unconfirmed_users

Remove unconfirmed user accounts? (Y/n)
Y
Invalid answer, enter option Y or n.
Remove unconfirmed user accounts? (Y/n)
Yes
Invalid answer, enter option Y or n.
Remove unconfirmed user accounts? (Y/n)
y
Invalid answer, enter option Y or n.
Remove unconfirmed user accounts? (Y/n)
n
Invalid answer, enter option Y or n.
Remove unconfirmed user accounts? (Y/n)
y
Invalid answer, enter option Y or n.

scinote@d23f4311f1e8:/usr/src/app$ rake data:clean

Remove expired temporary files? (Y/n)
y
Invalid answer, enter option Y or n.
Remove expired temporary files? (Y/n)
Y
Invalid answer, enter option Y or n.
Remove expired temporary files? (Y/n)
Y
Invalid answer, enter option Y or n.
Remove expired temporary files? (Y/n)

**Comment by Ducz0r** [comment]
It seems you've discovered a bug in our Rake tasks :smile:. We'll add it to our bug tracker and resolve it by the next release.

Anyway, if you need to do a "yes", simply press enter without entering any characters, and it should work.

**Comment by umardraz** [comment]
Ok here is the output

scinote@d23f4311f1e8:/usr/src/app$ rake data:clean

Remove expired temporary files? (Y/n)

I, [2016-06-03T16:18:32.964415 #53]  INFO -- : Cleaning temporary files older than 3 days
D, [2016-06-03T16:18:33.006003 #53] DEBUG -- :   TempFile Load (1.9ms)  SELECT "temp_files".* FROM "temp_files" WHERE (created_at < '2016-05-31 16:18:32.972515')

Remove unconfirmed user accounts? (Y/n)

I, [2016-06-03T16:18:35.899466 #53]  INFO -- : Cleaning unconfirmed users older than 3 days
D, [2016-06-03T16:18:35.907116 #53] DEBUG -- :   User Load (3.0ms)  SELECT "users".* FROM "users" WHERE (confirmed_at = NULL and created_at < '2016-05-31 16:18:35.899979')

Yesterday I had created few users and they are still non-confirmed but they were not deleted with the above command.

Br.

Umar

**Comment by Ducz0r** [comment]
How did you create users yesterday? If you created them with `rake db:add_user`, they should be automatically confirmed.

If you created them by using the "sign up" procedure, then they indeed need to be confirmed. For that to really work, you need to go into the running Docker container (`make cli`), and run the background jobs worker that will send these emails (`rake jobs:work`). Even if you don't have the email server configured, this command will output the contents of the sent emails into your console, and you can copy the confirmation link from there and enter it into the web browser to confirm the non-confirmed users.

If the above paragraph seems like too much trouble for you, the good thing with Rails is that it provides user with a very powerful console. So alternative to confirm users is to go into Docker container - `make cli`, followed by entering Rails console - `rails c`. Once inside the Rails console, you can find the users you wish to confirm (see [official Rails docs](http://guides.rubyonrails.org/active_record_querying.html)) by e.g. doing the following:
```ruby
u = User.find_by(email: "<user_email>")
u.update(confirmed_at: Time.now)
```

Hopefully that helps you,
Cheers

**Comment by umardraz** [comment]
Hello Thanks

But what if I want to delete them instead of confirm or anything else e.g

DELETE FROM "users" WHERE confirmed_at = NULL

Br.
Umar

**Comment by Ducz0r** [comment]
Simple
```ruby
User.where(confirmed_at: nil).destroy_all
```

**Comment by umardraz** [comment]
irb(main):012:0> User.where(confirmed_at: nil).destroy_all
  User Load (1.4ms)  SELECT "users".* FROM "users" WHERE "users"."confirmed_at" IS NULL
   (0.4ms)  BEGIN
  SQL (7.4ms)  DELETE FROM "users" WHERE "users"."id" = $1  [["id", 2]]
   (0.7ms)  ROLLBACK
ActiveRecord::InvalidForeignKey: PG::ForeignKeyViolation: ERROR:  update or delete on table "users" violates foreign key constraint "fk_rails_edec76c076" on table "organizations"
DETAIL:  Key (id)=(2) is still referenced from table "organizations".
: DELETE FROM "users" WHERE "users"."id" = $1

**Comment by Ducz0r** [comment]
Okay, seems that's not working.

Nonetheless, for sciNote managing purposes, the best option at the moment is to learn the Rails [ActiveRecords](http://guides.rubyonrails.org/active_record_querying.html) API and use console (`rails c`) to perform these operations.

**Comment by umardraz** [comment]
Hmm, 

Can we use ldap authentication with sciNote?

Br

Umar

**Comment by Ducz0r** [comment]
I'm sorry, but LDAP authentication is currently not supported. You can either develop it on your own, or else - we will definately consider integrating it if there is adequate feedback from the community.

**Comment by Ducz0r** [comment]
I'm closing this issue.
